examples/hnas/example_mlp.py

```diff
@@ -0,0 +1,228 @@
+            epoch=epoch + 1,
+            performance=float(curr_best),
+            eval_time=eval_time,
+            time_step=ts_now, **config)
+
+
+def callback(training_history, start_timestamp):
+    # This callback function will be executed every time AutoGluon collected some new data.
+    # In this example we will parse the training history into a .csv file and save it to disk, such that we can
+    # for example plot AutoGluon's performance during the optimization process.
+    # If you don't care about analyzing performance of AutoGluon online, you can also ignore this callback function and
+    # just save the training history after AutoGluon has finished.
+    import pandas as pd
+    task_dfs = []
+
+    # this function need to be changed if you return something else than accuracy
+    def compute_error(df):
+        return 1.0 - df["performance"]
+
+    def compute_runtime(df, start_timestamp):
+        return df["time_step"] - start_timestamp
+
+    for task_id in training_history:
+        task_df = pd.DataFrame(training_history[task_id])
+        task_df = task_df.assign(task_id=task_id,
+                                 runtime=compute_runtime(task_df, start_timestamp),
+                                 error=compute_error(task_df),
+                                 target_epoch=task_df["epoch"].iloc[-1])
+        task_dfs.append(task_df)
+
+    result = pd.concat(task_dfs, axis="index", ignore_index=True, sort=True)
+
+    # re-order by runtime
+    result = result.sort_values(by="runtime")
+
+    # calculate incumbent best -- the cumulative minimum of the error.
+    result = result.assign(best=result["error"].cummin())
+
+    result.to_csv("history.csv")
+
+
+# CLI
+def parse_args():
+    parser = argparse.ArgumentParser(description='runs autogluon on autoff benchmarks')
+
+    parser.add_argument('--num_trials', default=10, type=int,
+                        help='number of trial tasks')
+    parser.add_argument('--timeout', default=1000, type=int,
+                        help='runtime of autogluon in seconds')
+    parser.add_argument('--num_gpus', type=int, default=0,
+                        help='number of GPUs available to a given trial.')
+    parser.add_argument('--num_cpus', type=int, default=2,
+                        help='number of CPUs available to a given trial.')
+    parser.add_argument('--hostfile', type=argparse.FileType('r'))
+    parser.add_argument('--store_results_period', type=int, default=100,
+                        help='If specified, results are stored in intervals of '
+                             'this many seconds (they are always stored at '
+                             'the end)')
+    parser.add_argument('--scheduler', type=str, default='hyperband_promotion',
+                        choices=['hyperband_stopping', 'hyperband_promotion'],
+                        help='Asynchronous scheduler type. In case of doubt leave it to the default')
+    args = parser.parse_args()
```

Comment:
Can you check examples/hpo/mlp.py? We have quite a number of other args the user may want to change here.

Reply:
ok I've added a few more